docusaurus/src/components/Details.jsx

```diff
@@ -0,0 +1,46 @@
+import { useLocation } from "@docusaurus/router";
+import classNames from "classnames";
+import React from "react";
+import styles from "./Details.module.css";
+
+export const Details = ({ className, ...rest }) => {
```

Comment:
I tried testing with a nested `<details>` tag. 

While opening a new tab with a hash of a nested header worked, clicking on another header and then clicking on the header in that nested `<details>` section causes the entire parent details section to be collapsed. See recording below:

https://github.com/user-attachments/assets/6202181a-1750-46be-bd2a-d6ed62e54c1a

Ideally this would work as well, but doesn't necessarily need to block this merge.

Reply:
Nice catch! I'll create a new issue and prioritize it with Natalie :)